mozc packages not building

Foreword: I am the maintainer of `ibus-mozc` and `fcitx-mozc` packages.

[`ibus-mozc`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/inputmethods/ibus-engines/ibus-mozc/default.nix) and [`fcitx-mozc`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/inputmethods/fcitx-engines/fcitx-mozc/default.nix) packages are not building since the `2016-03-27 14:42:35` build.

It seems that the recent update to GCC5 via b07e7bfc7bc8f0b5428acaab026738866af32a28 triggered this build failure.

- [Hydra build](http://hydra.nixos.org/build/33690927)
- [log](http://hydra.nixos.org/build/33690927/nixlog/1)

It is complaining about a lot of undefined references to `protobuf`, here the tail of the log:

```
$ nix-build -A ibus-engines.mozc
```

```
gen/proto_out/protocol/user_dictionary_storage.pb.cc:(.text+0x4b4): undefined reference to `google::protobuf::internal::empty_string_'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o: In function `mozc::user_dictionary::UserDictionary_Entry::UserDictionary_Entry()':
gen/proto_out/protocol/user_dictionary_storage.pb.cc:(.text+0xb25): undefined reference to `google::protobuf::internal::empty_string_'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o: In function `mozc::user_dictionary::UserDictionary_Entry::SharedCtor()':
gen/proto_out/protocol/user_dictionary_storage.pb.cc:(.text+0xbee): undefined reference to `google::protobuf::internal::empty_string_'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o: In function `mozc::user_dictionary::UserDictionary_Entry::UserDictionary_Entry(mozc::user_dictionary::UserDictionary_Entry const&)':
gen/proto_out/protocol/user_dictionary_storage.pb.cc:(.text+0xcca): undefined reference to `google::protobuf::internal::empty_string_'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o: In function `mozc::user_dictionary::UserDictionary_Entry::MergeFrom(mozc::user_dictionary::UserDictionary_Entry const&)':
gen/proto_out/protocol/user_dictionary_storage.pb.cc:(.text+0xdb4): undefined reference to `google::protobuf::internal::empty_string_'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:gen/proto_out/protocol/user_dictionary_storage.pb.cc:(.text+0xe00): more undefined references to `google::protobuf::internal::empty_string_' follow
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0x20): undefined reference to `google::protobuf::Message::GetTypeName() const'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0x40): undefined reference to `google::protobuf::Message::InitializationErrorString() const'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0xd0): undefined reference to `google::protobuf::Message::GetTypeName() const'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0xf0): undefined reference to `google::protobuf::Message::InitializationErrorString() const'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0x180): undefined reference to `google::protobuf::Message::GetTypeName() const'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0x1a0): undefined reference to `google::protobuf::Message::InitializationErrorString() const'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0x230): undefined reference to `google::protobuf::Message::GetTypeName() const'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0x250): undefined reference to `google::protobuf::Message::InitializationErrorString() const'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0x2e0): undefined reference to `google::protobuf::Message::GetTypeName() const'
obj/protocol/gen/proto_out/protocol/user_dictionary_storage_proto.user_dictionary_storage.pb.o:(.data.rel.ro+0x300): undefined reference to `google::protobuf::Message::InitializationErrorString() const'
obj/client/client.client.o: In function `mozc::client::Client::DumpHistorySnapshot(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
../../client/client.cc:(.text+0x97b): undefined reference to `google::protobuf::Message::DebugString() const'
obj/ipc/ipc.ipc_path_manager.o: In function `mozc::IPCPathManager::CreateNewPathName()':
../../ipc/ipc_path_manager.cc:(.text+0x3df): undefined reference to `google::protobuf::internal::empty_string_'
obj/ipc/gen/proto_out/ipc/ipc_protocol.ipc.pb.o: In function `mozc::ipc::protobuf_AddDesc_ipc_2fipc_2eproto()':
gen/proto_out/ipc/ipc.pb.cc:(.text+0x286): undefined reference to `google::protobuf::internal::empty_string_'
obj/ipc/gen/proto_out/ipc/ipc_protocol.ipc.pb.o: In function `mozc::ipc::IPCPathInfo::IPCPathInfo()':
gen/proto_out/ipc/ipc.pb.cc:(.text+0x4aa): undefined reference to `google::protobuf::internal::empty_string_'
obj/ipc/gen/proto_out/ipc/ipc_protocol.ipc.pb.o: In function `mozc::ipc::IPCPathInfo::SharedCtor()':
gen/proto_out/ipc/ipc.pb.cc:(.text+0x573): undefined reference to `google::protobuf::internal::empty_string_'
obj/ipc/gen/proto_out/ipc/ipc_protocol.ipc.pb.o: In function `mozc::ipc::IPCPathInfo::IPCPathInfo(mozc::ipc::IPCPathInfo const&)':
gen/proto_out/ipc/ipc.pb.cc:(.text+0x64f): undefined reference to `google::protobuf::internal::empty_string_'
obj/ipc/gen/proto_out/ipc/ipc_protocol.ipc.pb.o:gen/proto_out/ipc/ipc.pb.cc:(.text+0x783): more undefined references to `google::protobuf::internal::empty_string_' follow
obj/ipc/gen/proto_out/ipc/ipc_protocol.ipc.pb.o:(.data.rel.ro+0x20): undefined reference to `google::protobuf::Message::GetTypeName() const'
obj/ipc/gen/proto_out/ipc/ipc_protocol.ipc.pb.o:(.data.rel.ro+0x40): undefined reference to `google::protobuf::Message::InitializationErrorString() const'
clang-3.7: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "src/build_mozc.py", line 1396, in <module>
    main()
  File "src/build_mozc.py", line 1392, in main
    procedure[1](cmd_opts, cmd_args, original_directory_name)
  File "src/build_mozc.py", line 1012, in BuildMain
    BuildOnLinux(options, targets, original_directory_name)
  File "src/build_mozc.py", line 968, in BuildOnLinux
    RunOrDie([make_command] + build_args + target_names)
  File "/tmp/nix-build-ibus-mozc-2.17.2313.102.drv-0/mozc-3306d3314499a54a4064b8b80bbc1bce3f6cfac4-src/src/build_tools/util.py", line 99, in RunOrDie
    '==========']))
build_tools.util.RunOrDieError: 
==========
 ERROR: ninja -C out_linux/Release ibus_mozc mozc_emacs_helper mozc_server mozc_tool mozc_renderer
==========
builder for ‘/nix/store/x9c44p4brgi675yn80ph778nx6cv514v-ibus-mozc-2.17.2313.102.drv’ failed with exit code 1
error: build of ‘/nix/store/x9c44p4brgi675yn80ph778nx6cv514v-ibus-mozc-2.17.2313.102.drv’ failed
```

Both mozc packages are using `clangStdenv` and `gyp` to build.

I tried a few things but couldn't fix the build.

@joachifm I saw you fixed many package not building in the recent days, maybe you know how to fix this issue?

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -1067,6 +1067,7 @@ in
 
     mozc = callPackage ../tools/inputmethods/ibus-engines/ibus-mozc {
       inherit (pythonPackages) gyp;
+      protobuf = protobuf.override { stdenv = clangStdenv; };
     };
 
     table = callPackage ../tools/inputmethods/ibus-engines/ibus-table {

